### DISCUSSION TITLE
Cant apply groupby dynamically on sqlalchmy orm query.

### DISCUSSION THREAD

**Discussion by arslanafzalch** [fastapi/fastapi#9792] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
query = session.query(func.count(), PatientInfo.tb_type)
query.group_by(PatientInfo.tb_type)
print(query)
return query
```

### Description

Hello Fastapi fellows, I have a script in which I have to apply filter and group_by dynamically in the latter stages if some specific conditions meet. The filter function is applying and working well, but group_by is not applying to the query. As in code, I gave a very basic example of what I am doing, but it's not working that way.

### Operating System

Windows

### FastAPI Version

fastapi-0.99.1

### Python Version

Python 3.10.10

### Additional Context

sqlalchemy-2.0.17
also try on sqlalchemy-1.4.46

**Comment by jgould22** [comment, marked as answer]
You probably want 
```python
query = session.query(func.count(), PatientInfo.tb_type)
query = query.group_by(PatientInfo.tb_type)
print(query)
return query
```

**Comment by ahmedabdou14** [comment]
Yes as answered by @jgould22 , sqlalchemy `query`  return the string representation of the final raw sql query that would be executed.

Meanwhile, I believe it is best to start a discussion on sqlalchemy repository for sqlalchemy related questions :)
